src/Configuration/Schema/SchemaConfigurationFactory.php

```diff
@@ -70,8 +70,8 @@ public function create(string $path, stdClass $rawConfig): SchemaConfiguration
     private static function createSource(stdClass $source): Source
     {
         return new Source(
-            (array) self::normalizeStringArray($source->directories ?? []),
-            (array) self::normalizeStringArray($source->excludes ?? [])
+            self::normalizeStringArray($source->directories ?? []),
+            self::normalizeStringArray($source->excludes ?? [])
```

Comment:
```diff
diff --git a/src/Configuration/Schema/SchemaConfigurationFactory.php b/src/Configuration/Schema/SchemaConfigurationFactory.php
index 4d554ce4..6f20f2c9 100644
--- a/src/Configuration/Schema/SchemaConfigurationFactory.php
+++ b/src/Configuration/Schema/SchemaConfigurationFactory.php
@@ -70,8 +70,8 @@ class SchemaConfigurationFactory
     private static function createSource(stdClass $source): Source
     {
         return new Source(
-            self::normalizeStringArray($source->directories ?? []),
-            self::normalizeStringArray($source->excludes ?? [])
+            isset($source->directories) ? self::normalizeStringArray($source->directories) : [],
+            isset($source->excludes) ? self::normalizeStringArray($source->excludes) : []
         );
     }
```

Why bother invoking a method when we already know we don't need to normalize an empty array?

Reply:
given the actual code, purely stylistic here and I find the change more expressive tbh